Sources/Amplitude/Utilities/PersistentStorageResponseHandler.swift

```diff
@@ -106,28 +108,32 @@ class PersistentStorageResponseHandler: ResponseHandler {
                 message: error
             )
             storage.remove(eventBlock: eventBlock)
-            return
+            return true
         }
         storage.splitBlock(eventBlock: eventBlock, events: events)
+        return true
     }
 
-    func handleTooManyRequestsResponse(data: [String: Any]) {
+    func handleTooManyRequestsResponse(data: [String: Any]) -> Bool {
         // wait for next time to pick it up
+        return false
     }
 
-    func handleTimeoutResponse(data: [String: Any]) {
+    func handleTimeoutResponse(data: [String: Any]) -> Bool {
         // Wait for next time to pick it up
+        return false
     }
 
-    func handleFailedResponse(data: [String: Any]) {
+    func handleFailedResponse(data: [String: Any]) -> Bool {
         // wait for next time to try again
+        return false
     }
 
-    func handle(result: Result<Int, Error>) {
+    func handle(result: Result<Int, Error>) -> Bool {
```

Comment:
considering all of the other functions are called from this, does it make sense to only have this function in the newer protocol?

Reply:
Good idea, this makes it much more concise.